### ISSUE TITLE
Stamp compiled assets with a checksum of :dependencies and on start up clean when different.

### ISSUE THREAD

**Issue by bhauman** [bhauman/lein-figwheel#413]
Stamp compiled assets with a checksum of :dependencies and perhaps :build-options on start up and clean when different. Before starting.

We can use closure defines or pre-amble to include it in the files.

Need to look at what ClojureScript is doing with stamping.

**Comment by TerjeNorderhaug** [comment]
This fix seem to have introduced a new issue: If multiple builds exists together, starting figwheel for a single build will clean out the other builds. For example, say there are two builds coexisting, "app" and "server": 

    lein do cljsbuild once app, cljsbuild once server

Starting figwheel for "app" will clean files built for the server (and vice versa):

    lein figwheel app

**Comment by bhauman** [comment]
I don't think that is what is causing the behavior your seeing ... but this deserves a look ...

**Comment by TerjeNorderhaug** [comment]
The purging of the previously builds seems to be accompanied by the message _Figwheel: Cleaning because dependencies changed_ which is output by `leiningen.figwheel/clean-on-dependency-change` when `on-stamp-change` finds there has been a dependency change. I don't know whether this is triggered by the changes in this patch or another recent one.

**Comment by bhauman** [comment]
@TerjeNorderhaug try "0.5.5-SNAPSHOT" this should be fixed

**Comment by TerjeNorderhaug** [comment]
I verified it's been fixed. Thanks.
